TestModels/Resource/runtimes/rust/tests/simple_resources_test.rs

```diff
@@ -22,13 +22,23 @@ async fn TestClient(config: SimpleResourcesConfig) {
 
 async fn TestNoneGetData(config: SimpleResourcesConfig, resource: SimpleResourceRef) {
     let input = allNone();
-    let result = resource.inner.borrow_mut().get_resource_data(input).unwrap();
+    let result = resource
+        .inner
+        .lock()
+        .unwrap()
```

Comment:
Not blocking on this PR, but I think we should really have resource operations defined on the resource ref structs as well, so customers doesn't have to mess with internal things. `borrow_mut()` wasn't great before in terms of encapsulation and now it's a bit worse.

Reply:
https://github.com/smithy-lang/smithy-dafny/issues/771